src/lein_monolith/dependency.clj

```diff
@@ -165,6 +165,74 @@
           (recur (conj result node)
                  (into (pop queue) (set/difference consumers result))))))))
 
+(defn unique-cycles
+  "Return a set of all unique cycles in dependency graph m."
+  [m]
+  {:pre [(map? m)]
+   :post [(set? %)]}
+  (let [path->cycles (fn path->cycles [path]
+                       {:pre [(seq path)]}
+                       (let [k (peek path)
+                             mks (set (keys m))
+                             vs (m k)
+                             path-set (set path)]
+                         (mapcat (fn [v]
+                                   (cond
+                                     ; found the cycle
+                                     (path-set v) [(into []
+                                                         ; drop non-cyclic prefix
+                                                         (drop-while (complement #{v}))
+                                                         (conj path v))]
+                                     :else (path->cycles
+                                             (conj path v))))
+                                 vs)))
+        all-cycles (mapcat #(path->cycles [%]) (keys m))
+        ; remove duplicate cycles (that involve the same deps)
+        [cycles-vecs _] (reduce (fn [[cycles-vecs cycle-sets] c]
```

Comment:
A simpler way to do this:
```clojure
(into #{}
      (map (comp first val))
      (group-by set all-cycles))
```

Reply:
Unfortunately that solution is slower because `group-by` is eager and traverses `all-cycles` an extra time. `all-cycles` might be quite large (it's every path through every cycle) and introducing a single small cycle (a likely case) may trigger many other larger cycles.

Reply:
I added a comment with your version to clarify the code.